mautic/mautic
# Campaign event execution not respecting configured restricted time window

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Functionality of "Schedule at" should same as "between the hours of" with combination of time delay.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a campaign
3. Add any campaign event
4. Add another campaign event with following configuration: 
     - Execute this event - at a relative time period
     - Delay of 30 minutes
     - Schedule at - 13:00

#### Expected result:
if (execution datetime of event created in step 2 + 30 minutes) is less than or equal to 13:00 for same date

event created in step 3 should be scheduled for 13:00 of same date
else

event created in step 3 should be scheduled for 13:00 of tomorrow
Note - if weekdays also selected then considered weekday also at time of schedule.
